### PR TITLE
Integrate the list command with sort options into RecipeIO and update UG + DG

### PR DESCRIPTION
**Pull request by PDHung1104**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -163,7 +165,7 @@ Use case ends.
 
 * *add [NAME, MINUTES, KCALS, ALLERGIES, CATEGORY, URL]* - This is the command a user can call to add a recipe. 
 * *help* - This shows the user all the available commands. 
-* *list* - This lists out a users recipebook for them. 
+* *list {SORT_TYPE}* - This lists out a users recipebook for them.
```
Comment (nidhi-nayak) [inline docs/DeveloperGuide.md:168]: let's keep this consistent with [] instead of {}

File: docs/UserGuide.md [hunk 2]
```diff
@@ -70,21 +70,23 @@ Deletes a recipe at a given recipe number. The application saves your recipe boo
...
-* A recipe's number can be found in the `list`, `find`, or `filter` commands.
+* A recipe's number can be found in the `list` (with no `SORT_TYPE` specifed), `find`, or `filter` commands.
 
 
 ### 4. Listing recipe book : `list`
 Lists a summarized version of your recipe book. This summarized version includes the
 recipes' name, date added, and url in the order they were added
 
-Format: `list`
+Format: `list {SORT_TYPE}`
```
Comment (nidhi-nayak) [inline docs/UserGuide.md:80]: no {} here

**Review by nidhi-nayak — CHANGES_REQUESTED** [review]
looks all good to me! just some super slight formatting, but other than that good to merge!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -139,6 +140,7 @@ How the component works:
 | v2.0    | foodie   | save and load recipes                      | share recipes with friends conveniently                       |
 | v2.0    | user     | find recipes without a certain ingredient  | save time finding recipes that I can eat                      |
 | v2.0    | user     | find a recipe by date                      | find a recipe if I don't remember its name                    |
+| v2.1 | user | see a my recipe list in a certain way | make decisions quicker |
```
[the remark this answers is not in the document]
Comment (PDHung1104) [reply docs/DeveloperGuide.md:143]: I think gh will automatically align the ```|```.

**Review by nidhi-nayak — APPROVED** [review]

**Pull request merged**